Having created an simple circuit. Trying to draw it with circuit.draw(output = 'mpl', interactive = True). The command produces some error as follows:
circuit.draw(output = 'mpl', interactive = True)
Traceback (most recent call last):
File "", line 1, in 
File "C:\Users\pc.conda\envs\qcircuit\lib\site-packages\qiskit\circuit\quantumcircuit.py", line 487, in draw
justify=justify)
File "C:\Users\pc.conda\envs\qcircuit\lib\site-packages\qiskit\visualization\circuit_visualization.py", line 218, in circuit_drawer
image.show()
File "C:\Users\pc\AppData\Roaming\Python\Python37\site-packages\matplotlib\figure.py", line 450, in show
manager.show()
File "C:\Users\pc\AppData\Roaming\Python\Python37\site-packages\matplotlib\backends_backend_tk.py", line 546, in show
self.canvas.tkcanvas.bind("", destroy)
File "C:\Users\pc.conda\envs\qcircuit\lib\tkinter_init.py", line 1251, in bind
return self._bind(('bind', self.w), sequence, func, add)
File "C:\Users\pc.conda\envs\qcircuit\lib\tkinter_init.py", line 1206, in _bind
self.tk.call(what + (sequence, cmd))
_tkinter.TclError: can't invoke "bind" command: application has been destroyed
Open anaconda prompt. Activate the environment created specifically for qiskit. Open python shell and type the following commands:
import qiskit
from qiskit import QuantumCircuit
circuit = QuantumCircuit(2,2)
circuit.h(0)
circuit.draw(output = 'mpl', interactive = True)
The error will show up.
I expect some interactive panel popping up like when I use plt.show() to show some matplotlib figures.
If this is not the correct way to get a mpl form of a circuit figure showing please tell me which reference should I be looking into.
Hi @skxsky I agree this is unexpected behaviour. This happens because we close the figure before returning it, to prevent it from rendering twice in Jupyter notebooks. If you run this code from a Jupyter notebook it should work, or you can save the image using
circuit.draw(output = 'mpl', filename="my_circuit.png"). I will try to have a look at getting the behaviour more inline with what is expected.
@maddy-tod I see. So rendering an interactive image in python shell is just not a feature. Apart from simply printing the circuit, are there any other ways to show a clearer image in python shell? If I were to use scripts then saving the image would work fine for me.
@skxsky no, printing the circuit or saving the image are the only options at the moment. I will try to get the interactive element fixed ASAP!
@maddy-tod Thanks, that solves my puzzles.
To solve this problem I did 'pip install pylatexenc' and it works now
